File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool solution(int x) {
    int temp = x;
    int sum = 0;
       
    while (temp >= 10){
        sum += temp % 10;
        temp /= 10;
    }        
    sum += temp;        
    return x % sum == 0;
}

File: 프로그래머스/lv1/12947. 하샤드 수/하샤드 수.java
class Solution {
    public boolean solution(int x) {
        String str = String.valueOf(x);
        String[] numberArr = str.split("");
        int sum = 0;

        for (int i = 0; i < numberArr.length; i++) {
            sum += Integer.parseInt(numberArr[i]);
        }

        if (x % sum == 0) {
            return true;
        } else {
            return false;
        }
    }
}